src/app/clients/add/AddClientForm.tsx

```diff
@@ -0,0 +1,237 @@
+    const [formErrors, setFormErrors] = useState(initialFormErrors);
+    const [submitError, setSubmitError] = useState(Errors.none);
+    const [submitErrorMessage, setSubmitErrorMessage] = useState("");
+    const [submitDisabled, setSubmitDisabled] = useState(false);
+
+    useEffect(() => {
+        if (fields.numberChildren <= fields.children.length) {
+            fieldSetter("children", fields.children.slice(0, fields.numberChildren));
+            return;
+        }
+
+        const extraChildren: Person[] = Array(fields.numberChildren - fields.children.length)
+            .fill(0)
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            .map((_item) => {
+                return {
+                    personType: "child",
+                    age: null,
+                };
+            });
+        fieldSetter("children", [...fields.children, ...extraChildren]);
+    }, [fields.numberChildren]); // eslint-disable-line react-hooks/exhaustive-deps
+
+    const fieldSetter = setField(setFields, fields);
+    const errorSetter = setError(setFormErrors, formErrors);
+
+    const submitForm = async (): Promise<void> => {
+        setSubmitDisabled(true);
+
+        const extraInformationWithNappy =
+            fields.nappySize === ""
+                ? fields.extraInformation
+                : `Nappy Size: ${fields.nappySize}, Extra Information: ${fields.extraInformation}`;
+
+        const clientRecord: ClientDatabaseInsertRecord = {
+            full_name: fields.fullName,
+            phone_number: fields.phoneNumber,
+            address_1: fields.addressLine1,
+            address_2: fields.addressLine2,
+            address_town: fields.addressTown,
+            address_county: fields.addressCounty,
+            address_postcode: fields.addressPostcode,
+            dietary_requirements: checkboxGroupToArray(fields.dietaryRequirements),
+            feminine_products: checkboxGroupToArray(fields.feminineProducts),
+            baby_food: fields.babyProducts,
+            pet_food: checkboxGroupToArray(fields.petFood),
+            other_items: checkboxGroupToArray(fields.otherItems),
+            delivery_instructions: fields.deliveryInstructions,
+            extra_information: extraInformationWithNappy,
+            signposting_call_required: fields.signpostingCall,
+            flagged_for_attention: fields.attentionFlag,
+        };
+
+        const inputError = checkErrorOnSubmit(formErrors, setFormErrors);
+        if (inputError) {
+            setSubmitError(Errors.submit);
+            setSubmitDisabled(false);
+            return;
+        }
+
+        try {
+            const ids = await insertClient(clientRecord);
+            try {
+                await insertFamily([...fields.adults, ...fields.children], ids.family_id);
+                router.push("/clients");
+            } catch (error) {
+                await deleteFailedInsert(ids.primary_key);
+                throw error;
```

Comment:
Just a quick note here that I ended up using a try inside a try because I prefer this over making `ids` a `let`.